yew/src/callback.rs

```diff
@@ -4,6 +4,61 @@ use std::cell::RefCell;
 use std::fmt;
 use std::rc::Rc;
 
+/// Flags that modify default callback behaviour
+#[derive(Eq, PartialEq, Clone, Copy, Hash, Debug)]
+pub struct Flags(u8);
+
+impl std::ops::BitAnd for Flags {
+    type Output = Self;
+
+    fn bitand(self, rhs: Self) -> Self {
+        Self(self.0 & rhs.0)
+    }
+}
+
+impl std::ops::BitAndAssign for Flags {
+    fn bitand_assign(&mut self, rhs: Self) {
+        self.0 &= rhs.0;
+    }
+}
+
+impl std::ops::BitOr for Flags {
+    type Output = Self;
+
+    fn bitor(self, rhs: Self) -> Self {
+        Self(self.0 | rhs.0)
+    }
+}
+
+impl std::ops::BitOrAssign for Flags {
+    fn bitor_assign(&mut self, rhs: Self) {
+        self.0 |= rhs.0;
+    }
+}
+
+impl Flags {
+    /// Return, if self hash f set inside of it
+    pub fn has_set(&self, f: Flags) -> bool {
+        *self & f != NO_FLAGS
+    }
+}
+
+/// Apply no Flags to callback. This is different from constructing a callback with None flags as it
+/// overrides the defaults for the callback type.
+pub const NO_FLAGS: Flags = Flags(0);
+/// Defines the event listener as passive.
+/// Yew sets sane defaults depending on the type of the listener.
+/// See [addEventListener](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEvent).
+pub const PASSIVE: Flags = Flags(1);
+/// Causes the event handler to not fire until the next animation frame
+/// Implies `PASSIVE`.
+// TODO: this flag can apply to Agents and Components as well
+pub const DEFER: Flags = Flags(1 << 1 | PASSIVE.0);
```

Comment:
For events that fire faster than the framerate of the of the page (like mousemove or very rapid keypresses). I found this to vastly improve performance, when I implemented draggable widgets.

Reply:
Nice, I think this can be solved with `RenderService` though

Reply:
Yes, but that is a lot of boilerplate. You need to:
1. Send the `Message` in the original callback.
2. Handle the `Message` on the `Component` by creating a `RenderTask` that sends a different `Message`.
3. Store a the `RenderTask` to prevent dropping.
4. Handle the new `Message` on the next animation frame.
5. Drop the `RenderTask`. 

Another reason to add this to callbacks is I want to later implement `DEFER` in the scheduler. Currently, if a component updates twice, it will generate 2 VNode trees and perform a diff against each, even if the frame is never rendered, because they happened in quick succession. If we defer component updates to the next animation frame, we can collapse multiple updates into one and save us a lot of allocations and node comparisons. As a bonus, users would be able to have their own callbacks executed only on the next animation frame by the scheduler. 

Further optimizations could be made by storing `VNode`s in a tree instead of the current Rc soup. This would allow to have a consistent view of all the nodes and their relations in an `App` and collapse all mutations on animation frame. To not have to traverse the entire tree for a diff, a component  can mark itself as dirty on update and mark all of its parents as having a dirty child. That way downwards traversal knows exactly which branches to go down for updates.